images/router/haproxy/conf/haproxy-config.template

```diff
@@ -624,6 +672,21 @@ backend {{ genBackendNamePrefix $cfg.TLSTermination }}:{{ $cfgIdx }}
           {{- end }}{{/* hsts header */}}
         {{- end }}{{/* is "edge" or "reencrypt" */}}
 
+          {{- range $idx, $http_request_header := $cfg.HTTPRequestHeaders }}
```

Comment:
There is a block above:
`{{- if matchValues (print $cfg.TLSTermination) "edge" "reencrypt" }}`, and I think the code that looks at headers needs to be within that block, not outside of it, since this does not apply to passthrough termination.

Reply:
@candita line https://github.com/openshift/router/blob/5e9cc11475d5c5b1b5fa31a1c0d069163a62cc10/images/router/haproxy/conf/haproxy-config.template#L528  already checks it.

Reply:
Lines 662 and 669 check it again, so it looks like a mistake that it isn't checked here.

Reply:
https://github.com/openshift/router/pull/438/files#diff-21bd64a4ad8a7927e9d59e99e09628efc884fe6d6291e50eb33c5deb3d2a79f0L684

is the end of the block which allows the non-edge or non-re-encrypt, edge and re-encrypt route except passthrough.

Reply:
Ok, I see lines 662 and 669 are checking for edge, reencrypt, whereas line 528 checks for none, edge, reencrypt.